src/services/expensesService.js

```diff
@@ -0,0 +1,56 @@
+let expenses = [];
+let nextId = 1;
+
+const resetExpenses = () => {
+  expenses = [];
+  nextId = 1;
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
+    .filter((expense) => !from || new Date(expense.spentAt) >= new Date(from))
+    .filter((expense) => !to || new Date(expense.spentAt) <= new Date(to));
+};
+
+const getExpenseById = (id) => {
+  return expenses.find((expense) => expense.id === +id);
+};
+
+const createExpense = (userId, spentAt, title, amount, category, note) => {
+  const expense = {
+    id: nextId++,
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expenses.push(expense);
+
+  return expense;
+};
+
+const updateExpense = (id, data) => {
+  const expense = getExpenseById(id);
+
+  Object.assign(expense, data);
+
+  return expense;
+};
+
+const deleteExpense = (id) => {
+  expenses = expenses.filter((expense) => expense.id !== +id);
```

Comment:
The 'deleteExpense' function modifies the 'expenses' array but does not return any value. This can be confusing for other developers who use this function and expect it to return a new array or the deleted expense. Consider returning a value that indicates the result of the operation.